### ISSUE TITLE
Create tool to generate stubs for an extension

### ISSUE THREAD

**Issue by AndrolGenhald** [vimeo/psalm#7513]
Now that extensions can be supported without being loaded at runtime, to fully support an extension we'll need complete stubs for it. To aid in this effort, it would be fantastic to have a tool to generate a complete stubfile for an extension.

Stubs will often need to be modified manually after generation for additional accuracy (for instance [`PDOStatement::fetch`](https://www.php.net/manual/en/pdostatement.fetch.php) should take `PDO::FETCH_*, PDO::FETCH_ORI_*, int` instead of `int, int, int`), but this tool should simplify the task significantly.

**Comment by orklah** [comment]
Yeah, the tool will probably have to be PHP version aware. Things like attributes must be retrieved but can't on older versions

**Comment by AndrolGenhald** [comment]
> Things like attributes must be retrieved but can't on older versions

I didn't even think of that, are there extensions using attributes already? It probably won't be a huge issue if some things like that are missing initially, the stubs can always be improved later.

**Comment by orklah** [comment]
> are there extensions using attributes already

I have no idea, but this may apply to other features. I can only think of attributes who we manage to make them do something almost retroactively (things like ReturnTypeWillChange) but things like promoted properties need specific methods in Reflection that are not available on older PHP version.

I wonder if this tool would not be better in a separate repository, we could actually use BetterReflection for better support on older php versions. I never had to use this project and even less with many php version, but we could probably cut off on specific code by using it as a polyfill

**Comment by AndrolGenhald** [comment]
> I wonder if this tool would not be better in a separate repository

I think one big advantage to having it in this repo is that it makes it as easy as possible for someone to generate stubs for us when they get a warning due to an unsupported extension. They'll already have psalm installed, so all they have to do is run it with the `--generate-extension-stub` flag.

A downside that I just thought of is that Psalm doesn't support PHP < 7.4 (on master, where this is targeted), so it'll be difficult to generate stubs for older versions.

**Comment by AndrolGenhald** [comment]
[This](https://github.com/sasezaki/php-extension-stub-generator) might do everything we need. I'll try playing around with it soon and see how it compares.
